Library/Homebrew/unpack_strategy/zip.rb

```diff
@@ -27,10 +27,17 @@ def self.can_extract?(path)
               .returns(SystemCommand::Result)
     }
     def extract_to_dir(unpack_dir, basename:, verbose:)
+      unzip = begin
+        Formula["unzip"]
+      rescue FormulaUnavailableError
+        nil
+      end
+
       with_env(TZ: "UTC") do
         quiet_flags = verbose ? [] : ["-qq"]
         result = system_command! "unzip",
                                  args:         [*quiet_flags, "-o", path, "-d", unpack_dir],
+                                 env:          { "PATH" => PATH.new(unzip&.opt_bin, ENV["PATH"]) },
```

Comment:
```suggestion
      unzip = which("unzip", ENV["HOMEBREW_PATH"])

      with_env(TZ: "UTC") do
        quiet_flags = verbose ? [] : ["-qq"]
        result = system_command! unzip,
                                 args:         [*quiet_flags, "-o", path, "-d", unpack_dir],
```
Any reason not to do this instead?

Reply:
Not sure. I am just copying what most other unpack strategy do.

Reply:
Avoiding an exception seems much nicer to me, but I suppose that can be done along with making similar changes to the other unpack strategies.

Reply:
I think it's better to rely on an `opt_bin`.